loading from `.ini` files API changes

edublancas opened the pull request:
## Describe your changes

* [API Change] When loading connections from a `.ini` file via `%sql --section section_name`, the section name is set as the connection alias
* [API Change] Starting connections from a `.ini` file via `%sql [section_name]` has been deprecated


Also:

* Added missing documentation about reading `.ini` files
* Added missing unit tests

## Issue number

Closes #X

## Checklist before requesting a review

- [ ] Performed a self-review of my code
- [ ] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [ ] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [ ] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)



<!-- readthedocs-preview jupysql start -->
----
:books: Documentation preview :books:: https://jupysql--803.org.readthedocs.build/en/803/

<!-- readthedocs-preview jupysql end -->

neelasha23 commented:
Can we display some message when running `%sql --section duck`? Right now it's not clear if it worked or not
<img width="648" alt="Screenshot 2023-08-16 at 3 30 33 PM" src="https://github.com/ploomber/jupysql/assets/29096135/b5c64c50-0067-4546-a947-3fcf1c578781">

edublancas commented on src/sql/parse.py:
```diff
@@ -2,29 +2,100 @@
+    parser = ConfigParser()
+
+    try:
+        cfg_content = Path(config.dsn_filename).read_text()
```
fixed https://github.com/ploomber/jupysql/pull/803/commits/683c3939dd5764ae3c1e4eedc39ac6c3cb5eeb6a

edublancas commented:
> 1. Can we display some message when running `%sql --section duck`? Right now it's not clear if it worked or not

good catch. this is unrelated to the `.ini` feature, opened: https://github.com/ploomber/jupysql/issues/807

> 2\. Entire stack trace displayed if duplicate sections present:

fixed and added unit tests https://github.com/ploomber/jupysql/pull/803/commits/ba6616edd49c4e02cefe02b85d57005421ac74f1 (also covers duplicate keys)

> After loading section duckdb if we run %sql [duck2] it doesn't throw any deprecation warning:

this is the default behavior for the warnings module, it'll only show the warning once

![image](https://github.com/ploomber/jupysql/assets/989250/a2f65b6c-6114-46ee-8d0e-37e32344cc65)
